python/lsst/qa/explorer/mkClassifier.py

```diff
@@ -0,0 +1,170 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Trains the star galaxy classifier
+"""
+
+import numpy as np
+from sklearn.ensemble import AdaBoostClassifier
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.calibration import CalibratedClassifierCV
+
+import lsst.pex.config as pexConfig
+from lsst.pipe.base import Task
+
+__all__ = ["MakeStarGalaxyClassifierConfig", "MakeStarGalaxyClassifierTask"]
+
+
+class MakeStarGalaxyClassifierConfig(pexConfig.Config):
+    """Config for making the star galaxy classifier
+
+    Notes
+    -----
+    The version name is used to save the classifier and is used by StarGalaxyClassifierTask to identify
+    the requested classifier.
+    """
+    version = pexConfig.Field(
+        doc="Version information",
+        dtype=str,
+        default="HSC-COSMOS-20180538",
+        )
+
+    classifierType = pexConfig.Field(
+        doc="Morphology only (morph), morphology and color only (color) or both (both)",
+        dtype=str,
+        default="both"
+        )
+
+    columnsMorph = pexConfig.ListField(
+        doc="Columns for the morphology only classifier",
+        dtype=str,
+        default=["magPsf-magModelHSC-R", "magPsf-magModelHSC-I", "magPsf-magModelHSC-Z", "m", "b",
+                 "seeingHSC-G", "seeingHSC-R", "seeingHSC-I", "seeingHSC-Z", "seeingHSC-Y",
+                 "fluxSNHSC-G", "fluxSNHSC-R", "fluxSNHSC-I", "fluxSNHSC-Z", "fluxSNHSC-Y",
+                 "magPsf-magModelErrHSC-R", "magPsf-magModelErrHSC-I", "magPsf-magModelErrHSC-Z"]
+        )
+
+    columnsColor = pexConfig.ListField(
+        doc="Columns for the morphology and color based classifier",
+        dtype=str,
+        default=["HSC-G-HSC-R", "HSC-R-HSC-I", "HSC-I-HSC-Z", "HSC-Z-HSC-Y", "magPsf-magModelHSC-G",
+                 "magPsf-magModelHSC-R", "magPsf-magModelHSC-I", "magPsf-magModelHSC-Z",
+                 "magPsf-magModelHSC-Y", "m", "b", "fluxSNHSC-G", "fluxSNHSC-R", "fluxSNHSC-I",
+                 "fluxSNHSC-Z", "fluxSNHSC-Y", "seeingHSC-G", "seeingHSC-R", "seeingHSC-I", "seeingHSC-Z",
+                 "seeingHSC-Y", "magPsf-magModelErrHSC-G", "magPsf-magModelErrHSC-R",
+                 "magPsf-magModelErrHSC-I", "magPsf-magModelErrHSC-Z", "magPsf-magModelErrHSC-Y"]
+        )
+
+    filters = pexConfig.ListField(
+        doc="Filters used in the classifier should be given in wavelength order",
+        dtype=str,
+        default=["HSC-G", "HSC-R", "HSC-I", "HSC-Z", "HSC-Y"]
+        )
+
+
+class MakeStarGalaxyClassifierTask(Task):
+    """Train the star galaxy classifier on a pre made training set and save the classifier
```

Comment:
You may  want to add to the task doc string what algorithm is being used for the classifier. I know you mention it in the run method but just looking at the task name and task doc string seems to imply that task is more generic that it is.